### PR TITLE
remove escape percent signs from select_sql

### PR DESCRIPTION
**Pull request by dgoo2308**
For table and column names, % should not be written as %% in a select statement.

this results in an error when a column or table has a `%` character in the name.

```
INFO syncing vw_SkiveMonthlyStatistic full table
{"type": "SCHEMA", "stream":  ..... "SURFACE_BAD_%":  ......., "key_properties": []}
{"type": "ACTIVATE_VERSION", "stream": "dbo-vw_SkiveMonthlyStatistic", "version": 1627788239240}
INFO METRIC: {"type": "timer", "metric": "job_duration", "value": 0.3792839050292969, "tags": {"job_type": "sync_table", "database": "dbo", "table": "vw_SkiveMonthlyStatistic", "status": "failed"}}
CRITICAL (207, b"Invalid column name 'SURFACE_BAD_%%'.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n")
```

### CODE REVIEW

File: tap_mssql/sync_strategies/common.py [hunk 1]
```diff
@@ -84,12 +84,7 @@ def generate_select_sql(catalog_entry, columns):
     escaped_columns = [escape(c) for c in columns]
 
     select_sql = "SELECT {} FROM {}.{}".format(",".join(escaped_columns), escaped_db, escaped_table)
-
-    # escape percent signs
-    select_sql = select_sql.replace("%", "%%")
-    return select_sql
-
-
+
```
Comment (wmattern0) [inline tap_mssql/sync_strategies/common.py:87]: you got rid of the return statement, line 90.  That should be kept.

**Review by wmattern0 — CHANGES_REQUESTED** [review]
return statement

**Comment by wmattern0** [comment]
Was able to run this code against a MSSQL db with percent signs in column names after adding back the return stmt.  Running using the code in master was getting the same error as described in the original PR comment.

**Comment by s7clarke10** [comment]
As a FYI, the escaping of the % has come from the original mysql code.

https://github.com/transferwise/pipelinewise-tap-mysql/blob/f08a0a1f450b826c07544af77e402ef33ff4ad62/tap_mysql/sync_strategies/common.py#L87

Having read this change, I can see why it has been requested. I do wonder if this line to escape the % is removed, whether it will have other side effects, perhaps mysql just required the % sign to be escaped because it was getting confused with Python variable substitution?

File: tap_mssql/sync_strategies/common.py (hunk 1 above)
Comment (dgoo2308) [reply tap_mssql/sync_strategies/common.py:87]: @wmattern0 good catch

**Comment by dgoo2308** [comment]
> As a FYI, the escaping of the % has come from the original mysql code.
> 
> https://github.com/transferwise/pipelinewise-tap-mysql/blob/f08a0a1f450b826c07544af77e402ef33ff4ad62/tap_mysql/sync_strategies/common.py#L87
> 
> Having read this change, I can see why it has been requested. I do wonder if this line to escape the % is removed, whether it will have other side effects, perhaps mysql just required the % sign to be escaped because it was getting confused with Python variable substitution?

indeed this seems to be ported from mysq, which needed the % to be escaped to %%, whereas here the MSSQL colomn names are escape with double quotes in stead of back tick in the mysql-tap.

It's clear that that with the double quotes the % does not need to be replaced by %% in double quoted strings for MSSQL.

**Comment by dgoo2308** [comment]
@wintersrd could you consider to add this to the V2?

**Comment by wmattern0** [comment]
One issue that I found with using this tap is sending the singer records to target-snowflake([pipelinewise](https://github.com/transferwise/pipelinewise-target-snowflake.git#2.2.0)) causes a python error.  

At some point it does string formatting on the column names and the '%' in the column name is read as a variable, not as a literal.  Rather than moving forward with this tap and target combo, I am giving up and using a different pattern.

**Comment by s7clarke10** [comment]
@dgoo2308 , this PR has been merged into main under this change: https://github.com/wintersrd/pipelinewise-tap-mssql/pull/39 . Are you happy for this PR to be closed as it has been actioned?

**Comment by s7clarke10** [comment]
Closing Pull Request as this has been actioned : https://github.com/wintersrd/pipelinewise-tap-mssql/pull/39

**Pull request closed without merging**